### PR TITLE
V 1 1 lp whitelist

### PR DESCRIPTION
**Pull request by asardon**

### CODE REVIEW

File: contracts/interfaces/IBasePool_v_1_1.sol [hunk 1]
```diff
@@ -81,10 +81,9 @@ interface IBasePool_v_1_1 {
         uint256 liquidityBnd1,
         uint256 liquidityBnd2
     );
-
+    event LpWhitelistUpdate(address indexed maxLoanPerColl, bool isApproved);
```
Comment (jpick713) [inline contracts/interfaces/IBasePool_v_1_1.sol:84]: odd name for the address, but it technically works!

File: test/rpl-usdc-testing.js [hunk 2]
```diff
@@ -116,23 +116,46 @@ describe("RPL-USDC Pool Testing", function () {
...
-    await pool.connect(lp1).addLiquidity(lp1.address, addAmount, timestamp+60, 0);
-    
+    const blocknum = await ethers.provider.getBlockNumber()
+    const timestamp = (await ethers.provider.getBlock(blocknum)).timestamp
+    // check adding only possible if on lpwhitelist
+    await expect(pool.connect(lp1).addLiquidity(lp1.address, addAmount, timestamp+60, 0)).to.be.revertedWithCustomError(pool, "UnapprovedSender")
+    await expect(pool.connect(lp1).toggleLpWhitelist(lp1.address)).to.be.revertedWithCustomError(pool, "UnapprovedSender")
+    await expect(pool.connect(lp2).toggleLpWhitelist(lp1.address)).to.be.revertedWithCustomError(pool, "UnapprovedSender")
+    await pool.connect(deployer).toggleLpWhitelist(lp1.address)
+    await pool.connect(lp1).addLiquidity(lp1.address, addAmount, timestamp+60, 0)
```
Comment (jpick713) [inline test/rpl-usdc-testing.js:135]: should be fine, but maybe add test case after you are whitelisted that you still can't add on someone else's behalf
`await expect(pool.connect(lp1).addLiquidity(lp2.address, addAmount, timestamp+60, 0)).to.be.revertedWithCustomError(pool, "UnapprovedSender")`

File: test/rpl-usdc-testing.js [hunk 3]
```diff
@@ -116,23 +116,46 @@ describe("RPL-USDC Pool Testing", function () {
...
+    await expect(pool.connect(lp2).claimCreator()).to.be.revertedWithCustomError(pool, "UnapprovedSender")
+    await pool.connect(newCreator).claimCreator()
+
+    // check lpwhitelisting only possible for new creator
+    await expect(pool.connect(deployer).toggleLpWhitelist(lp2.address)).to.be.revertedWithCustomError(pool, "UnapprovedSender")
+    await expect(pool.connect(lp2).addLiquidity(lp2.address, addAmount, timestamp+60, 0)).to.be.revertedWithCustomError(pool, "UnapprovedSender")
+    await pool.connect(newCreator).toggleLpWhitelist(lp2.address)
+    await pool.connect(lp2).addLiquidity(lp2.address, addAmount, timestamp+60, 0)
+    await pool.connect(newCreator).toggleLpWhitelist(lp2.address)
+    await expect(pool.connect(lp2).addLiquidity(lp2.address, addAmount, timestamp+60, 0)).to.be.revertedWithCustomError(pool, "UnapprovedSender")
```
Comment (jpick713) [inline test/rpl-usdc-testing.js:150]: nice, looks pretty thorough on all the cases for newCreator

**Review by jpick713 — APPROVED** [review]

File: contracts/BasePool_v_1_1.sol [hunk 4]
```diff
@@ -145,11 +146,10 @@ abstract contract BasePool_v_1_1 is IBasePool_v_1_1 {
         uint256 _referralCode
     ) external override {
         // verify LP info and eligibility
+        if (msg.sender != _onBehalfOf || !lpWhitelist[msg.sender]) {
+            revert();
+        }
```
[the remark this answers is not in the document]
Comment (asardon) [reply contracts/BasePool_v_1_1.sol:151]: exactly, won't need peripherals to add for our poc case

File: test/rpl-usdc-testing.js (hunk 2 above)
Comment (asardon) [reply test/rpl-usdc-testing.js:135]: good catch, adding now

File: contracts/interfaces/IBasePool_v_1_1.sol (hunk 1 above)
Comment (asardon) [reply contracts/interfaces/IBasePool_v_1_1.sol:84]: good catch, fixed

File: contracts/pools/rpl-usdc/PoolRplUsdc_v_1_1.sol [hunk 5]
```diff
@@ -43,7 +43,7 @@ contract PoolRplUsdc_v_1_1 is BasePool_v_1_1 {
         uint256 _liquidityBnd2
     ) external {
         if (msg.sender != poolCreator) {
-            revert();
+            revert UnapprovedSender();
         }
         if (_maxLoanPerColl == 0) revert InvalidMaxLoanPerColl();
         if (_r1 <= _r2 || _r2 == 0) revert InvalidRateParams();
```
[the remark this answers is not in the document]
Comment (asardon) [reply contracts/pools/rpl-usdc/PoolRplUsdc_v_1_1.sol:49]: yeah agree, maybe for v2, but as you said, would do the simple +1 parameterization for now if needed

**Review by jpick713 — APPROVED** [review]

**Pull request merged**